test-data/unit/fine-grained-modules.test

```diff
@@ -1080,18 +1115,64 @@ import a
 [file a.py]
 import b
 b.foo()
+
 [file a.py.2]
 import b
 b.foo(0)
 [file b.py.2]
 def foo() -> None: pass
+
+[stale a, b]
+
 [file b.py.3]
 def foo(x: int) -> None: pass
+
+[stale2 b]
+
 [out]
 ==
 a.py:2: error: Too many arguments for "foo"
 ==
 
+[case testRenameAndDeleteModuleAfterCache-skip-nocache]
+import a
+[file a.py]
+from b1 import f
+f()
+[file b1.py]
+def f() -> None: pass
+[file b2.py.2]
+def f() -> None: pass
+[delete b1.py.2]
+[file a.py.2]
+from b2 import f
+f()
+
+-- in no cache mode, there is no way to know about b1 yet
+[stale a, b2]
+
+[out]
+==
+
+[case testDeleteModuleAfterCache-skip-nocache]
+import a
+[file a.py]
+from b import f
+f()
+[file b.py]
+def f() -> None: pass
+[delete b.py.2]
+
+-- in no cache mode, there is no way to know about b yet,
+-- but a should still get triggered
+[stale a]
```

Comment:
I don't fully understand why `a` is included here, since only `b` had file changes (it was deleted). Please explain in the comment why `a` is triggered.

Reply:
The initial cache mode run picks up changes to files that are not "fresh", in order to pick up modules with changed dependencies.